js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -2073,42 +2075,50 @@ const EditingController = modules.ViewController.inherit((function() {
             const oldEditRowIndex = that._getVisibleEditRowIndex();
 
             if(!isRowEditMode(that)) {
-                result = Deferred();
-                this.executeOperation(result, () => {
-                    this._closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData);
-                    result.resolve();
+                const deferred = new Deferred();
+                result = new Deferred();
+                this.executeOperation(deferred, () => {
+                    this._closeEditCellCore(isError, oldEditRowIndex, withoutSaveEditData).always(result.resolve);
```

Comment:
`closeEditCellCore` can not be rejected, why are you using `always` instead of `done`?

Reply:
I think it does not matter in this particular case. This call is analogue of the this.closeEditCell(null, isNewRow).always in the deleteRow method. Especially if we add the case when _closeEditCellCore will reject the returned promise in future.

Reply:
> Especially if we add the case when _closeEditCellCore will reject the returned promise in future.

Tbh, resolving `closeEditCell`'s deferred when `_closeEditCellCore` rejects it's own looks strange. But if you think it doesn't matter - OK.